src/messaging/ExchangeContext.cpp

```diff
@@ -608,6 +646,18 @@ CHIP_ERROR ExchangeContext::HandleMessage(uint32_t messageCounter, const Payload
         return CHIP_ERROR_INCORRECT_STATE;
     }
 
+#if CONFIG_DEVICE_LAYER && CHIP_CONFIG_ENABLE_ICD_SERVER
+    DeviceLayer::ChipDeviceEvent event;
+    event.Type                                  = DeviceLayer::DeviceEventType::kChipMsgRxEventHandled;
+    event.RxEventContext.wasReceived            = true;
```

Comment:
So here we are going to kick things along in active mode on receipt of non-response messages, right?  Why?  Why do we not treat a duplicate or ack in the same way?  What is the spec basis for this behavior?

Reply:
So based on our Slack conversation, I opened a spec Issue to add more details in the spec about the reasons that would require to be in active mode.
https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/7295